src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = new Node<>(null, value, null);
+        } else {
+            if (head.next == null) {
+                tail = new Node<>(head, value, null);
+                head.next = tail;
+            } else {
+                Node<T> newNode = new Node<>(tail, value, null);
+                tail.next = newNode;
+                tail = newNode;
+            }
+        }
+        size++;
+
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIsIndexExist(index, size, "Invalid index");
+
+        if (index == size) {
+            add(value);
+            return;
+        }
 
+        if (index == 0) {
+            addAtBeginning(value);
+        } else {
+            addByIndex(value, index);
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
```

Comment:
for better performance it's better to iterate our list depending on index if it's between start and middle - begin from the start in other case begin from an end